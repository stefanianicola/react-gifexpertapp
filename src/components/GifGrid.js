import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {

    const { data: images, loading } = useFetchGifs(category);


    return (
        <>
            <h3>{category}</h3>

            {
                loading ? (<p>Cargando...</p>) : (
                    <div className='card-grid'>

                        <ol>
                            {
                                images.map(img => {
                                    return <GifGridItem
                                        key={img.id}
                                        {...img} />
                                })
                            }
                        </ol>
                    </div>
                )
            }

        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
