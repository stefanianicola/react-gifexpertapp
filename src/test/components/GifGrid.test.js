import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('pruebas en GifGrid', () => {
    const category = 'Dragon Ball';

    it('que se cree correctamente el componente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('debe mostrar items cuando se cargan img', () => {

        const imgs = [{
            id: 'aaa',
            url: 'https://hhh',
            title: 'algo'
        }];

        useFetchGifs.mockReturnValue({
            data: imgs,
            loading: false
        });
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(imgs.length);

    });


})