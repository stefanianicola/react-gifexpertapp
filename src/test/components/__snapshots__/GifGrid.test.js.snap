// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifGrid debe mostrar items cuando se cargan img 1`] = `
<Fragment>
  <h3>
    Dragon Ball
  </h3>
  <div
    className="card-grid"
  >
    <ol>
      <GifGridItem
        id="aaa"
        key="aaa"
        title="algo"
        url="https://hhh"
      />
    </ol>
  </div>
</Fragment>
`;

exports[`pruebas en GifGrid que se cree correctamente el componente 1`] = `
<Fragment>
  <h3>
    Dragon Ball
  </h3>
  <p>
    Cargando...
  </p>
</Fragment>
`;
